pkg/apiserver/user/auth.go

```diff
@@ -55,23 +56,26 @@ type TokenResponse struct {
 }
 
 func (f *authenticateForm) Authenticate(tidbForwarder *tidb.Forwarder) (*utils.SessionUser, error) {
-	// TODO: Support non TiDB auth
 	if !f.IsTiDBAuth {
-		return nil, ErrSignInUnsupportedAuthType.New("unsupported auth type, only TiDB auth is supported")
-	}
-	db, err := tidbForwarder.OpenTiDB(f.Username, f.Password)
-	if err != nil {
-		if errorx.Cast(err) == nil {
+		err := globalUtil.VerifyKvModeAuthKey(tidbForwarder.EtcdClient, f.Password)
```

Comment:
you should not use the EtcdClient inside the tidbForwarder, it's the forwarder's own impl detail and should not be exposed. Better to accept another parameter instead.

Reply:
Got it